### DISCUSSION TITLE
docs/cctv/apps/ispy

### DISCUSSION THREAD

**Discussion by giscus** [AMD-NICK/blog.amd-nick.me#32] [Docs comments]
# docs/cctv/apps/ispy

ispy agent main view

https://blog.amd-nick.me/docs/cctv/apps/ispy

**Comment by lalik77** [comment]
Привет! Спасибо за статью. Расскажи, пожалуйста, как у тебя реализовано подключение камер. У меня iSpy на ноутбуке поднят в локальной сети, но планирую установить видеонаблюдение на даче с интернетом через сим-карту, где IP адрес динамический. Ты использовал статический IP? У меня есть идея синхронизировать записи с облаком, подобно твоему варианту на выделенном сервере. Однако, я думаю, что из-за отсутствия публичного IP, мне может быть сложно скопировать твой опыт. Встроенная опция отправки в облако требует бизнес-подписки. Возможно, установить клиента, который синхронизирует записи с облачными сервисами? Что думаешь?

**Reply by AMD-NICK** [reply]
Привет. Если я правильно понял, то на даче у тебя нет проводного интернета, но есть мобильный (с динамическим, серым IP). Также на даче стоит ноутбук с установленным ispy, к которому подключены камеры. Тебе нужно, чтобы ты удаленно мог открывать ispy, который установлен на том ноутбуке, верно? Тоесть, по сути, задача состоит в том, чтобы подключиться к ноутбуку?

Если я понял правильно, то решением может быть использование Tailscale. Если камеры свободно "общаются" с ноутбуком, то скорее всего (!), достаточно будет установить на него Tailscale, затем в админ панели Tailscale переименовать ноутбук в какой-нибудь "cctv_notebook". После этого с любого устройства, на котором будет включен Tailscale по идее можно будет подключиться к ISPY через `http://cctv_notebook:8090`.

Подключиться можно будет только если будет включен Tailscale с одной учетной записью на обеих устройствах. Например, на ноутбуке на даче и на ПК дома. Т.е. никто другой, кто не авторизирован в Tailscale с твоей учетной записью не сможет подключиться к ispy

**Reply by lalik77** [reply]
Да, все верно понял. Единственное, мне не нужно подключаться к ноутбуку, так как у камер есть свое приложение и слот для карты памяти. Следовательно, удаленный просмотр возможен, и даже производители предлагают облачное хранилище за дополнительную плату. Мне важно, чтобы в зимний период, когда нас там нет, система работала с облачным хранилищем. То есть, в случае срабатывания триггера, запись автоматически отправлялась в облако или на сервер через FTP, например.

**Reply by AMD-NICK** [reply]
В таком случае на ум приходит только написание какого-нибудь скрипта, который будет регулярно проверять директорию, появились ли там новые файлы, затем отправлять их куда-нибудь. Можно попробовать использовать что-то вроде rclone (https://hub.docker.com/r/rclone/rclone), но я еще не работал с ним

**Reply by lalik77** [reply]
Спасибо за ответ, я буду смотреть в сторону облачных сервисов.

**Reply by lalik77** [reply]
Привет , решил сегодня установить Tailscale а в России нельзя думаю из за санкций. Скажи пожалуйста есть ли возможность обойти или нет ? <br>
<br>
![image](https://github.com/AMD-NICK/blog.amd-nick.me/assets/45859153/834d3bb8-8808-4317-8c47-6a495174eb74)
<br><br>
На моем маке тоже не установить и на Андроиде тоже. Если есть возможномть обойти  , подскажи пожалуйста в какую сторону смотреть или не стоит время терять?

Дополню , решил реализовать через netbird.io но ребята с ispy предусмотрели такую возможность и защитили , так что вариант через vpn не работает. С облаком у меня получилось .

![2024-01-07_16-19-33](https://github.com/AMD-NICK/blog.amd-nick.me/assets/45859153/22f61220-d4b7-4838-a33b-5efefebc270f)

**Reply by AMD-NICK** [reply]
>  подскажи пожалуйста в какую сторону смотреть

Когда я пытался установить на свой сервер в РУ, то тоже столкнулся с такой проблемой. Решил проблему скачиванием .deb файла напрямую через curl/wget, затем `sudo dpkg -i file.deb`

**Comment by calutiya** [comment]
Благодарю за инструкцию 🙏
Скажи пожалуйста, ты не заметил? Когда делаются фото то они делаются как бы из substream, а не из основного кадра? Подскажи пожалуйста, как это побороть?

**Reply by calutiya** [reply]
Я нашёл :) в настройке "Photos" нужно указывать путь к качественным фоткам. Подсмотреть его можно в базе данных камер ispyconnect: https://www.ispyconnect.com/camera/dahua - да, у меня Dahua, но там уйма производителей. И тогда фотки делаются прям полноразмерные!

**Reply by AMD-NICK** [reply]
Если делать из основного кадра, то вроде оверлея от ИИ не будет (квадратика с обозначением объекта на фото)
